app/views/articles/_search.html.erb

```diff
@@ -45,68 +43,64 @@
           return;
         }
         var filters = e.target.dataset.filter;
-        window.history.replaceState(null, null, "/search?q=" + query + "&filters=" + filters);
+        window.history.replaceState(null, null, "/search?search_fields=" + query + "&filters=" + filters);
         var className = e.target.className;
         for (var i = 0; i < filterButts.length; i++) {
           filterButts[i].classList.remove("selected");
         }
         if (className.indexOf("selected") == -1) {
           e.target.classList.add("selected");
-          window.history.replaceState(null, null, "/search?q=" + query + "&filters=" + filters);
-          search(query, index, filters);
+          window.history.replaceState(null, null, "/search?search_fields=" + query + "&filters=" + filters);
+          search(query, filters);
         } else {
-          window.history.replaceState(null, null, "/search?q=" + query);
-          search(query, index, "");
+          window.history.replaceState(null, null, "/search?search_fields=" + query);
+          search(query, "");
         }
       }
     }
   }
 
-  function search(query, index, filters) {
-    var hashtags = query.match(/#\w+/g)
-    var searchObj = {
-      hitsPerPage: 60,
-      page: "0",
-      queryType: "prefixNone",
-      attributesToRetrieve: [
-        'id',
-        'title',
-        'path',
-        'class_name',
-        'comments_count',
-        'tag_list',
-        'readable_publish_date',
-        'positive_reactions_count',
-        'flare_tag',
-        'user',
-        'reading_time'
-      ],
-      attributesToHighlight: [],
-      exactOnSingleWordQuery: "none",
-      attributesToSnippet: ['body_text:19', 'comments_blob:8'],
-    }
+  function search(query, filters) {
+    var hashtags = query.match(/#\w+/g);
+    var searchTerm = query.replace(/#/g, '').trim();
+    var searchHash = { per_page: 60, page: 0 };
 
-    if (filters === "MY_POSTS" && userData()) {
-      searchObj["tagFilters"] = [['user_' + userData()['id']]]
-    } else if (hashtags != null && hashtags.length > 0) {
+    if (hashtags != null && hashtags.length > 0) {
```

Comment:
This will check if it's `null` or `undefined`. I'm assuming the value would only ever be `null`, `undefined` or an array.

```suggestion
    if (hashtags && hashtags.length > 0) {
```

Reply:
I would assume so, that code I just left from before but I'm game for an easy refactor.